src/router/chainable_router.rs

```diff
@@ -0,0 +1,71 @@
+use hyper::method::Method;
+use middleware::Middleware;
+use nickel::Nickel;
+use router::http_router::{HttpRouter};
+
+pub struct ChainableRoute<'a, D: Sync + Send + 'static = ()> {
+    path: String,
```

Comment:
I think this can be `&'a str`, or you might need to introduce a new lifetime `'b` if the `route` chaining is to work (haven't tried)